packages/main/src/StepInput.js

```diff
@@ -0,0 +1,540 @@
+		 *
+		 * @type {String}
+		 * @since 1.0.0-rc.9
+		 * @private
+		 * @defaultvalue ""
+		 */
+		ariaLabel: {
+			type: String,
+		},
+
+		/**
+		 * Receives id(or many ids) of the elements that label the <code>ui5-step-input</code>.
+		 *
+		 * @type {String}
+		 * @defaultvalue ""
+		 * @private
+		 * @since 1.0.0-rc.9
+		 */
+		ariaLabelledby: {
+			type: String,
+			defaultValue: "",
+		},
+
+		_decIconDisabled: {
+			type: Boolean,
+			noAttribute: true,
+		},
+
+		_incIconDisabled: {
+			type: Boolean,
+			noAttribute: true,
+		},
+
+		_focused: {
+			type: Boolean,
+			noAttribute: true,
+		},
+
+		_inputFocused: {
+			type: Boolean,
+			noAttribute: true,
+		},
+
+		_previousValue: {
+			type: Float,
+			noAttribute: true,
+		},
+
+	},
+	slots: /** @lends sap.ui.webcomponents.main.StepInput.prototype */ {
+		/**
+		 * Defines the value state message that will be displayed as pop up under the <code>ui5-step-input</code>.
+		 * <br><br>
+		 *
+		 * <b>Note:</b> If not specified, a default text (in the respective language) will be displayed.
+		 * <br>
+		 * <b>Note:</b> The <code>valueStateMessage</code> would be displayed,
+		 * when the <code>ui5-step-input</code> is in <code>Information</code>, <code>Warning</code> or <code>Error</code> value state.
+		 * @type {HTMLElement}
+		 * @since 1.0.0-rc.7
+		 * @slot
+		 * @public
+		 */
+		valueStateMessage: {
+			type: HTMLElement,
+		},
+	},
+	events: /** @lends sap.ui.webcomponents.main.StepInput.prototype */ {
+		/**
+		 * Fired when the input operation has finished by pressing Enter or on focusout.
+		 *
+		 * @event
+		 * @public
+		*/
+		change: {},
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ *
+ * <h3>Usage</h3>
+ *
+ * For the <code>ui5-step-input</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/StepInput.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.StepInput
+ * @extends UI5Element
+ * @tagname ui5-step-input
+ * @public
+ */
+class StepInput extends UI5Element {
+
+	constructor() {
+		super();
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents");
+	}
+
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return StepInputCss;
+	}
+
+	static get template() {
+		return StepInputTemplate;
+	}
+
+	static get dependencies() {
+		return [];
+	}
+
+	static async onDefine() {
+		await Promise.resolve([
+			fetchI18nBundle("@ui5/webcomponents"),
+		]);
+	}
+
+	get decIconTitle() {
+		return this.i18nBundle.getText(STEPINPUT_DEC_ICON_TITLE);
+	}
+
+	get decIconName() {
+		return "less";
+	}
+
+	get incIconTitle() {
+		return this.i18nBundle.getText(STEPINPUT_INC_ICON_TITLE);
+	}
+
+	get incIconName() {
+		return "add";
+	}
+
+	get type() {
+		return InputType.Number;
+	}
+
+	get _placeholder() {
+		return this.placeholder + " ";
```

Comment:
this is a bit unusual, what is the reason for that

Reply:
This was because of rendering glitch - when the input (and placeholder) is right-aligned, the last letter of the placeholder was cut partially because it is italic. I have found CSS solution for this issue, and this getter is removed.